File: leetcode/medium/0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        r=len(height)-1
        max_water=0
        while l<r:
            c=min(height[l],height[r])*(r-l)
            max_water=max(max_water,c)

            if height[l]<height[r]:
                l+=1
            else:
                r-=1
        return max_water
        
        

File: leetcode/medium/0011-container-with-most-water/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0;
        int r=height.size()-1;
        int max_water=0;
        
        while (l<r){
            int c=std::min(height[l], height[r]) * (r - l);
            max_water=std::max(max_water,c);
            if (height[l]<height[r]){
                l++;
            } else {
                r--;
            }
        }
        return max_water;   
    }
};